### PR TITLE
Enable BGP validates content Collection

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>

### CODE REVIEW

File: roles/bgp_health/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+- debug:
+    var: hostvars
+
+- name: Parse bgp summary
+  ansible.utils.cli_parse:
+    command: "show bgp summary"
+    parser:
+      name: ansible.netcommon.pyats
```
Comment (NilashishC) [inline roles/bgp_health/tasks/main.yml:9]: The `show bgp summary` isn't the most appropriate command in all cases and is deprecated for IOS-XE.

We should (a) find the correct command(s) for all platforms and map them in vars, (b) evaluate which parsing engine is best suited for converting the output from those commands to structured data.
In quite a few cases, using native parsing engine and our own hand written parsers turn out to be more precise.

File: roles/bgp_health/includes/health_checks/junos.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+- set_fact:
```
Comment (NilashishC) [inline roles/bgp_health/includes/health_checks/junos.yaml:1]: We don't seem to be evaluating any op state data here.

File: roles/bgp_health/includes/write_to_file.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  set_fact:
+    inventory_directory: ./inventory
+  when: inventory_directory is not defined
+
+- name: Create inventory directory
+  file:
+    path: "{{ inventory_directory }}/host_vars/{{ inventory_hostname }}"
+    state: directory
+
+- name: Write each resource to a file
```
Comment (NilashishC) [inline roles/bgp_health/includes/write_to_file.yaml:12]: What is the benefit of persisting operational state data to a file?

File: roles/bgp_health/includes/health_checks/junos.yaml (hunk 2 above)
Comment (rohitthakur2590) [reply roles/bgp_health/includes/health_checks/junos.yaml:1]: At moment we are extracting limited  information as it is still in progress.
There will be  a generic model with more attributes for example all_neighbours_up/down/active, etc

File: roles/bgp_health/includes/write_to_file.yaml (hunk 3 above)
Comment (rohitthakur2590) [reply roles/bgp_health/includes/write_to_file.yaml:12]: nothing as such , it is to be removed/may be put as optional.

File: roles/bgp_health/tasks/main.yml (hunk 1 above)
Comment (rohitthakur2590) [reply roles/bgp_health/tasks/main.yml:9]: yes you are absolutly right, This work is already in progress and will be updated shortly.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,79 @@
...
+- This role enables users to create a runtime brownfield inventory with all the BGP configuration in terms of host vars. These host vars are ansible facts which have been gathered through the *_bgp_global and *_bgp_address_family network resource module.The tasks offered by this role could be observed as below:
+
+#### Perform BGP Health Checks
+```yaml
+health_checks.yml
+---
+- hosts: ios
+  gather_facts: false
+  tasks:
+  - name: BGP Manager
```
Comment (trishnaguha) [inline README.md:26]: We should specify what this task is actually doing. One liner on health check.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,79 @@
...
+                min_count: 1
+```
+
+
+#### Building Brownfield Inventory
+```
+- hosts: ios
+  gather_facts: false
+  tasks:
+  - name: BGP Manager
```
Comment (trishnaguha) [inline README.md:47]: We should specify what this task is actually doing.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,79 @@
...
+      actions:
+        - name: persist
+```
+
+#### Gather BGP Facts
+```
+- hosts: ios
+  gather_facts: false
+  tasks:
+  - name: BGP Manager
```
Comment (trishnaguha) [inline README.md:60]: We should specify what this task is actually doing.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,79 @@
...
+      actions:
+        - name: gather
+```
+
+#### Gather Deploy BGP Configuration
+```
+- hosts: ios
+  gather_facts: false
+  tasks:
+  - name: BGP Manager
```
Comment (trishnaguha) [inline README.md:73]: We should specify what this task is actually doing.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

**Pull request merged**